src/Algebra/Local/Structures.agda

```diff
@@ -0,0 +1,60 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Local algebraic structures
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Algebra.Local.Structures where
+
+open import Level using (Level; _⊔_) renaming (suc to lsuc)
+open import Data.Product using (∃-syntax; _×_)
+open import Algebra.Core using (Op₁; Op₂)
+open import Algebra.Definitions using (Invertible)
+open import Algebra.Structures using (IsCommutativeRing)
+open import Relation.Binary.Core using (Rel)
+open import Relation.Binary.Structures using (IsEquivalence; IsApartness)
+open import Relation.Binary.Definitions using (Tight)
+import Relation.Binary.Properties.Apartness as A
+open import Relation.Nullary using (¬_)
+
+
+module _
+  {c ℓ₁ ℓ₂} {Carrier : Set c}
+  (_≈_ : Rel Carrier ℓ₁)
+  (_#_ : Rel Carrier ℓ₂)
+  (_+_ _*_ : Op₂ Carrier) (-_ : Op₁ Carrier) (0# 1# : Carrier)
+  where
+
+  -- private to avoid clash with Monoid._-_
+  private
+    _-_ : Op₂ Carrier
+    x - y = x + (- y)
```

Comment:
This definition of `_-_` should already be available in `IsCommutativeRing` inherited from `IsGroup`. That would allow us to do the following:
```agda
  record IsLocalCommutativeRing : Set (c ⊔ ℓ₁ ⊔ ℓ₂) where
    field
      isCommutativeRing : IsCommutativeRing _≈_ _+_ _*_ -_ 0# 1#
      isApartness       : IsApartness _≈_ _#_
    
    open IsCommutativeRing isCommutativeRing public
    open IsApartness isApartness public

    field
      #⇔invertible      : ∀ {x y} → Iso (x # y) (Invertible _≈_ 1# _*_ (x - y))
```
If it's not available, then it really should be and we've messed up the transitive imports in the main algebra hierarchy somewhere. If you're not up to finding where, let me know and I'll try and find the place where it goes wrong.

Reply:
That worked! I thought imports were not allowed interleaved with field declarations. It's a nice trick.